src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,651 @@
+		return Constant.messages.getString(MESSAGE_PREFIX+ "name");
+	}
+
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+   
+	@Override
+	public int getCweId() {
+		return CWED_ID;
+	}
+
+	@Override
+	public int getWascId() {
+		return WASC_ID;
+	}
+	
+	
+    /**
+     * Scan for NoSQL injection vulnerabilities
+     * @inheritDoc
+     *
+     */
+	public void scan(HttpMessage msg, String param, String value) {
+
+		String paramInjected, valueInjected, attackType;
+		HttpMessage injectedMsg, baseMsg;
+		long intervalBaseMessage, intervalInjectedMessage; 
+		Instant now;
+		boolean onlyExactMatch;
+
+		try {
+			
+			baseMsg = getNewMsg();
+			setParameter(baseMsg, param, TOKEN);
+
+			now = Instant.now();
+			sendAndReceive(baseMsg);
+			intervalBaseMessage = ChronoUnit.MILLIS.between(now, Instant.now());
+			
+			
+			for(NOSQLDB nosql: NOSQLDB.values()) {
+
+				if(inScope(nosql.getDbTech())){
+
+					
+					/**
+					 * TODO check for the URL_ENCEODED format
+					 * 
+					 */
+					if (true) { 
+
+						for(String[] pv : nosql.getUrlEncParamValueInjection()) {
+							
+							onlyExactMatch=false;
+							attackType = PARVAL_ATTACK;
+							injectedMsg = getNewMsg();
+							paramInjected = param+pv[0];
+							valueInjected = pv[1];
+							
+							setParameter(injectedMsg, paramInjected, valueInjected);
+		
+							try {
+								sendAndReceive(injectedMsg, false);
+
+								if(isBingo(baseMsg, injectedMsg, paramInjected, valueInjected, attackType, nosql, 
+										onlyExactMatch)) {
+									break;
+								}
+		
+							} catch (SocketException ex) {
+		
+								if (log.isDebugEnabled()) 
+									log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+											" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}
+							
+						}
+						
+						
+						for(String v : nosql.getUrlEncValueInjection()) {
+							
+							onlyExactMatch=false;
+							attackType = VALUE_ATTACK;
+							injectedMsg = getNewMsg();
+							valueInjected = v;
+							setParameter(injectedMsg, param, valueInjected);
+							
+							try {
+								sendAndReceive(injectedMsg, false);
+								
+								if(isBingo(baseMsg, injectedMsg, param, valueInjected, attackType, nosql, 
+										onlyExactMatch)) {
+									break;
+								}
+		
+							} catch (SocketException ex) {
+		
+								log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 

```

Comment:
Anytime there's concatenation in `log.debug` is should be wrapped in an `isDebugEnabled` check